### PR TITLE
[notifications][docs] rename and export `getExpoPushTokenAsync` parameter type

### PR DESCRIPTION
**Pull request by Simek**
# Why

Refs: https://exponent-internal.slack.com/archives/C1QMWQ1P0/p1675723296628999

# How

After switching Notifications package to leverage the autogenerated docs one of the methods parameters types was missing. This PR renames and exports the problematic param type to make the documentation complete.

# Test Plan

The updated docs data file was tested locally, by running docs app.

# Preview

Screenshot 2023-02-07 at 11 45 01

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by kbrandwijk** [comment]
I guess the fact that this type wasn't exported before makes this not a breaking change.

**Comment by Simek** [comment]
Generally, changing types is not considered by us as a breaking change. In this case, since the type was not exported earlier,  it is not possible that this change would affect users. It would be possible to break some users code if the type was exported locally, and not on the root level (some people might use the build file import directly to obtain the type information), but this fortunately was not the case.

File: packages/expo-notifications/src/Tokens.types.ts [hunk 1]
```diff
@@ -62,3 +62,53 @@ export interface ExpoPushToken {
...
+   * You may want to customize that if you don't want to install `expo-application` and still use the sandbox APNs.
+   * @platform ios
+   */
+  development?: boolean;
+  /**
+   * @deprecated Use `projectId` instead.
+   * The ID of the experience to which the token should be attributed.
+   * Defaults to [`Constants.manifest.id`](/constants/#constantsmanifest) exposed by `expo-constants`.
+   * When building with EAS Build, or in the bare workflow, **this is required** and you must provide a value which takes the shape `@username/projectSlug`,
+   * where `username` is the Expo account that the project is associated with, and `projectSlug` is your [`slug`](/config/app.mdx#slug) from Expo config.
```
Comment (amandeepmittal) [inline packages/expo-notifications/src/Tokens.types.ts:95]: Can we modify this by moving the deprecated label below the description such that the description of the field is not included in the deprecation notice.

Update:

```ts
 /**
   * The ID of the experience to which the token should be attributed.
   * Defaults to [`Constants.manifest.id`](/constants/#constantsmanifest) exposed by `expo-constants`.
   * When building with EAS Build, or in the bare workflow, **this is required** and you must provide a value which takes the shape `@username/projectSlug`,
   * where `username` is the Expo account that the project is associated with, and `projectSlug` is your [`slug`](/config/app.mdx#slug) from Expo config.
   *
   * @deprecated Use `projectId` instead.
   */
  experienceId?: string;
```

End result:

CleanShot 2023-02-07 at 17 50 58@2x

Comment (Simek) [reply packages/expo-notifications/src/Tokens.types.ts:95]: Sure, that was for sure not intended. Feel free to update this PR branch if you want.

**Comment by amandeepmittal** [comment]
Updated the docs as per the latest conversation in the Slack thread with Will and Simek. Let me know your feedback on this.

cc: @Simek @kbrandwijk

**Comment by brentvatne** [comment]
i'm about to cut sdk 48 docs, can you cherry-pick this to sdk-48 and ensure it's in sdk 48 docs when this ready?

**Review by amandeepmittal — APPROVED** [review]

**Pull request merged**